modules/core/src/utils/datafile.cpp

```diff
@@ -17,6 +17,10 @@
 
 #include <opencv2/core/utils/configuration.private.hpp>
 
+// Include private header file to prevent `-Wundef` to raise a warning
+// on undefined `OPENCV_HAVE_FILESYSTEM_SUPPORT`.
+#include "opencv2/core/utils/filesystem.private.hpp"
```

Comment:
IMHO, this comment is redundant here.

Reply:
I'll remove all instances of this comments.